File: Same_Tree/answer.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:

    def isSameTree(self, p, q):
        if p is None and q is None:
            return True
        elif p is None or q is None:
            return False
        elif p.val != q.val:
            return False
        else:
            return self.isSameTree(q.left, p.left) and self.isSameTree(q.right, p.right)

File: Invert_Binary_Tree/answer.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {TreeNode}
def invert_tree(root)
    return root if root.nil?

    temp = root.left
    root.left = root.right
    root.right = temp

    invert_tree(root.left) unless root.left.nil?
    invert_tree(root.right) unless root.right.nil?

    root
end